teuthology/suite/run.py

```diff
@@ -196,15 +196,58 @@ def choose_ceph_version(self, ceph_hash):
             log.info('skipping ceph package verification')
 
     def choose_teuthology_branch(self):
+        """Select teuthology branch, check if it is present in repo and
+        return the branch name value.
+
+        The branch name value is determined in the following order:
+
+        Use ``--teuthology-branch`` argument value if supplied.
+
+        Use ``TEUTH_BRANCH`` environment variable value if declared.
+
+        If yaml file ``qa/.teuthology`` can be found in suite repo
+        supplied with ``--suite-repo`` or ``--suite-dir`` and
+        has ``teuthology_branch`` record then use its value.
+
+        Use ``teuthology_branch`` value if it is set in the one
+        of the teuthology config files ``$HOME/teuthology.yaml``
```

Comment:
looks like ```$HOME/teuthology.yaml``` and ```/etc/teuthology.yaml``` aren't being read in this version of the patch

Reply:
It is done at 246 line with this code:
```
check_branch = config.get('teuthology_branch', None)
```

Reply:
ah, got it thanks